jbpm-flow/src/main/java/org/jbpm/process/instance/timer/TimerManager.java

```diff
@@ -261,9 +261,10 @@ public void deserialize(MarshallerReaderContext inCtx, Timer timer) throws Class
                     long now = pctx.getKnowledgeRuntime().getSessionClock().getCurrentTime();
                     // overdue timer                    
                     if (then < now) {
-                        trigger = new OverdueTrigger(trigger, pctx.getKnowledgeRuntime());
+                        trigger = new OverdueTrigger(trigger);
                     }
                 }
+                trigger.initialize(pctx.getWorkingMemory().getKnowledgeRuntime());
```

Comment:
Current version is getting `kruntime` from `pctx.getKnowledgeRuntime()` method. So not really sure why we are now retrieving in a different way by using `pctx.getWorkingMemory().getKnowledgeRuntime()`?
If `kruntime` is not an instance of `InternalWorkingMemory` this will return a null and thus it might fail later on - this wasn't previous/current behavior.
Can you please clarify?